In [1]:
import pandas as pd

In [2]:
data = pd.read_csv('retail supermarket.csv')

## ***Question No.1***

### 1- Find record if Sales greater than 500 and Quantity greater than 5?

## ***Answer No.1***

In [3]:
data[(data['Sales'] > 500) & (data['Quantity'] > 5)]

Ship Mode    Segment        Country           City         State  \
7     Standard Class   Consumer  United States    Los Angeles    California   
10    Standard Class   Consumer  United States    Los Angeles    California   
16    Standard Class   Consumer  United States        Madison     Wisconsin   
27    Standard Class   Consumer  United States   Philadelphia  Pennsylvania   
35       First Class  Corporate  United States     Richardson         Texas   
...              ...        ...            ...            ...           ...   
9879  Standard Class   Consumer  United States  New York City      New York   
9925     First Class  Corporate  United States  New York City      New York   
9942  Standard Class   Consumer  United States        Anaheim    California   
9947    Second Class  Corporate  United States   Indianapolis       Indiana   
9948    Second Class  Corporate  United States   Indianapolis       Indiana   

      Postal Code   Region         Category Sub-Category     Sales  Quantity  \
7           90032     West       Technology       Phones   907.152         6   
10          90032     West        Furniture       Tables  1706.184         9   
16          53711  Central  Office Supplies      Storage   665.880         6   
27          19140     East        Furniture    Bookcases  3083.430         7   
35          75080  Central       Technology       Phones  1097.544         7   
...           ...      ...              ...          ...       ...       ...   
9879        10035     East        Furniture       Chairs   573.174         7   
9925        10035     East  Office Supplies      Binders  1087.936         8   
9942        92804     West  Office Supplies      Storage   998.820         9   
9947        46203  Central        Furniture       Chairs  1925.880         6   
9948        46203  Central  Office Supplies   Appliances  2405.200         8   

      Discount     Profit  
7          0.2    90.7152  
10         0.2    85.3092  
16         0.0    13.3176  
27         0.5 -1665.0522  
35         0.2   123.4737  
...        ...        ...  
9879       0.1    63.6860  
9925       0.2   353.5792  
9942       0.0    29.9646  
9947       0.0   539.2464  
9948       0.0   793.7160  

[431 rows x 13 columns]

####  There are 431 records with sales exceeding 500 and quantity exceeding 5.

## ***Question No.2***

### 2- Find records if Discount is more than 0.5 (50%) or Profit is negative.

## ***Answer No.2***

In [4]:
data[(data['Discount'] > 0.5) | (data['Profit'] < 1)]

Ship Mode      Segment        Country             City  \
3     Standard Class     Consumer  United States  Fort Lauderdale   
14    Standard Class  Home Office  United States       Fort Worth   
15    Standard Class  Home Office  United States       Fort Worth   
23      Second Class     Consumer  United States     Philadelphia   
27    Standard Class     Consumer  United States     Philadelphia   
...              ...          ...            ...              ...   
9931  Standard Class     Consumer  United States   San Bernardino   
9937    Second Class    Corporate  United States      Los Angeles   
9944    Second Class  Home Office  United States          Seattle   
9962     First Class  Home Office  United States          Houston   
9977  Standard Class  Home Office  United States      Los Angeles   

             State  Postal Code   Region         Category Sub-Category  \
3          Florida        33311    South        Furniture       Tables   
14           Texas        76106  Central  Office Supplies   Appliances   
15           Texas        76106  Central  Office Supplies      Binders   
23    Pennsylvania        19140     East        Furniture       Chairs   
27    Pennsylvania        19140     East        Furniture    Bookcases   
...            ...          ...      ...              ...          ...   
9931    California        92404     West        Furniture    Bookcases   
9937    California        90049     West        Furniture       Tables   
9944    Washington        98103     West  Office Supplies      Storage   
9962         Texas        77041  Central        Furniture    Bookcases   
9977    California        90032     West  Office Supplies    Fasteners   

          Sales  Quantity  Discount     Profit  
3      957.5775         5      0.45  -383.0310  
14      68.8100         5      0.80  -123.8580  
15       2.5440         3      0.80    -3.8160  
23      71.3720         2      0.30    -1.0196  
27    3083.4300         7      0.50 -1665.0522  
...         ...       ...       ...        ...  
9931   683.3320         4      0.15   -40.1960  
9937    71.0880         2      0.20    -1.7772  
9944    40.7400         3      0.00     0.4074  
9962   383.4656         4      0.32   -67.6704  
9977    13.8600         7      0.00     0.0000  

[2202 rows x 13 columns]

####  There are 2202 records with a discount greater than 50% or negative profit.

## ***Question No.3***

### 3- Display Ship Mode and Profit columns if region is East using loc

## ***Answer No.3***

In [5]:
data.loc[data['Region'] == 'East', ['Ship Mode','Profit']]

Ship Mode     Profit
23      Second Class    -1.0196
27    Standard Class -1665.0522
28    Standard Class    -7.0532
29    Standard Class    15.5250
30    Standard Class     1.1016
...              ...        ...
9968  Standard Class   331.1910
9969  Standard Class     6.5975
9981     First Class     5.3690
9984  Standard Class    15.1200
9985  Standard Class    16.1240

[2848 rows x 2 columns]

####  There are 2848 records showing Ship Mode and Profit for the East region.

## ***Question No.4***

### 4- How many records belong to the "Corporate" segment and the "Central" region?

## ***Answer No.4***

In [6]:
data[(data['Segment'] == 'Corporate') & (data['Region'] == 'Central')]

Ship Mode    Segment        Country          City     State  \
21    Standard Class  Corporate  United States       Fremont  Nebraska   
22    Standard Class  Corporate  United States       Fremont  Nebraska   
35       First Class  Corporate  United States    Richardson     Texas   
36       First Class  Corporate  United States    Richardson     Texas   
41    Standard Class  Corporate  United States    Naperville  Illinois   
...              ...        ...            ...           ...       ...   
9947    Second Class  Corporate  United States  Indianapolis   Indiana   
9948    Second Class  Corporate  United States  Indianapolis   Indiana   
9949    Second Class  Corporate  United States  Indianapolis   Indiana   
9950    Second Class  Corporate  United States  Indianapolis   Indiana   
9951    Second Class  Corporate  United States  Indianapolis   Indiana   

      Postal Code   Region         Category Sub-Category     Sales  Quantity  \
21          68025  Central  Office Supplies          Art    19.460         7   
22          68025  Central  Office Supplies   Appliances    60.340         7   
35          75080  Central       Technology       Phones  1097.544         7   
36          75080  Central        Furniture  Furnishings   190.920         5   
41          60540  Central       Technology       Phones   147.168         4   
...           ...      ...              ...          ...       ...       ...   
9947        46203  Central        Furniture       Chairs  1925.880         6   
9948        46203  Central  Office Supplies   Appliances  2405.200         8   
9949        46203  Central       Technology  Accessories    83.970         3   
9950        46203  Central       Technology  Accessories    39.890         1   
9951        46203  Central  Office Supplies      Binders    17.380         2   

      Discount    Profit  
21         0.0    5.0596  
22         0.0   15.6884  
35         0.2  123.4737  
36         0.6 -147.9630  
41         0.2   16.5564  
...        ...       ...  
9947       0.0  539.2464  
9948       0.0  793.7160  
9949       0.0   15.9543  
9950       0.0   14.7593  
9951       0.0    8.6900  

[673 rows x 13 columns]

####  There are 673 records for Corporate segment in the Central region.

## ***Question No.5***

### 5- Find all orders in the "Furniture" category but not in "Chairs" sub-category.

## ***Answer No.5***

In [7]:
data[(data['Category'] == 'Furniture') & ~(data['Sub-Category'] == 'Chairs')]

Ship Mode      Segment        Country             City       State  \
0       Second Class     Consumer  United States        Henderson    Kentucky   
3     Standard Class     Consumer  United States  Fort Lauderdale     Florida   
5     Standard Class     Consumer  United States      Los Angeles  California   
10    Standard Class     Consumer  United States      Los Angeles  California   
24    Standard Class     Consumer  United States             Orem        Utah   
...              ...          ...            ...              ...         ...   
9962     First Class  Home Office  United States          Houston       Texas   
9964    Second Class    Corporate  United States           Newark    Delaware   
9980    Second Class     Consumer  United States        Lafayette   Louisiana   
9989    Second Class     Consumer  United States            Miami     Florida   
9990  Standard Class     Consumer  United States       Costa Mesa  California   

      Postal Code   Region   Category Sub-Category      Sales  Quantity  \
0           42420    South  Furniture    Bookcases   261.9600         2   
3           33311    South  Furniture       Tables   957.5775         5   
5           90032     West  Furniture  Furnishings    48.8600         7   
10          90032     West  Furniture       Tables  1706.1840         9   
24          84057     West  Furniture       Tables  1044.6300         3   
...           ...      ...        ...          ...        ...       ...   
9962        77041  Central  Furniture    Bookcases   383.4656         4   
9964        19711     East  Furniture  Furnishings    13.4000         1   
9980        70506    South  Furniture       Tables    85.9800         1   
9989        33180    South  Furniture  Furnishings    25.2480         3   
9990        92627     West  Furniture  Furnishings    91.9600         2   

      Discount    Profit  
0         0.00   41.9136  
3         0.45 -383.0310  
5         0.00   14.1694  
10        0.20   85.3092  
24        0.00  240.2649  
...        ...       ...  
9962      0.32  -67.6704  
9964      0.00    6.4320  
9980      0.00   22.3548  
9989      0.20    4.1028  
9990      0.00   15.6332  

[1504 rows x 13 columns]

####  There are 1504 orders in the Furniture category that are not in the Chairs sub-category.

## ***Question No.6***

### 6- Which orders have discount (>0.3) and still earned profit (>0)?

## ***Answer No.6***

In [8]:
data[(data['Discount'] > 0.3) & (data['Profit'] > 0)]

Ship Mode      Segment        Country           City         State  \
343     Second Class    Corporate  United States   Philadelphia  Pennsylvania   
614     Second Class     Consumer  United States     Grove City          Ohio   
986   Standard Class  Home Office  United States       Pasadena         Texas   
1233     First Class    Corporate  United States     Cincinnati          Ohio   
1679  Standard Class    Corporate  United States   Philadelphia  Pennsylvania   
1912    Second Class    Corporate  United States        Houston         Texas   
1986     First Class     Consumer  United States       Columbus          Ohio   
2125  Standard Class  Home Office  United States   Philadelphia  Pennsylvania   
2135  Standard Class     Consumer  United States   Philadelphia  Pennsylvania   
2728     First Class     Consumer  United States         Toledo          Ohio   
3781    Second Class     Consumer  United States   Philadelphia  Pennsylvania   
4003  Standard Class     Consumer  United States    San Antonio         Texas   
4491    Second Class     Consumer  United States       Columbus          Ohio   
4812    Second Class     Consumer  United States   Philadelphia  Pennsylvania   
5384  Standard Class     Consumer  United States   Philadelphia  Pennsylvania   
5385  Standard Class     Consumer  United States   Philadelphia  Pennsylvania   
6425  Standard Class     Consumer  United States   Philadelphia  Pennsylvania   
6951  Standard Class    Corporate  United States         Dallas         Texas   
7042  Standard Class     Consumer  United States   Philadelphia  Pennsylvania   
8123  Standard Class     Consumer  United States   Philadelphia  Pennsylvania   
8799     First Class     Consumer  United States   Philadelphia  Pennsylvania   
8808  Standard Class     Consumer  United States   Philadelphia  Pennsylvania   
9106  Standard Class     Consumer  United States        Houston         Texas   
9516    Second Class     Consumer  United States  Bowling Green          Ohio   
9617  Standard Class    Corporate  United States  Bowling Green          Ohio   
9798  Standard Class    Corporate  United States         Toledo          Ohio   

      Postal Code   Region    Category Sub-Category     Sales  Quantity  \
343         19140     East  Technology       Phones    71.928        12   
614         43123     East  Technology       Phones   119.940        10   
986         77506  Central  Technology     Machines   336.510         3   
1233        45231     East  Technology      Copiers   839.988         2   
1679        19134     East  Technology       Phones    32.382         3   
1912        77041  Central  Technology     Machines   597.132         3   
1986        43229     East  Technology      Copiers  1439.976         4   
2125        19143     East  Technology       Phones    62.958         7   
2135        19134     East  Technology       Phones    94.920         4   
2728        43615     East  Technology       Phones   629.958         7   
3781        19120     East  Technology       Phones   269.982         3   
4003        78207  Central  Technology     Machines   287.910         3   
4491        43229     East  Technology      Copiers   659.988         2   
4812        19134     East  Technology      Copiers  1199.976         4   
5384        19143     East  Technology       Phones   358.200         3   
5385        19143     East  Technology       Phones   545.916        14   
6425        19120     East  Technology      Copiers  8399.976         4   
6951        75217  Central  Technology     Machines   479.988         2   
7042        19134     East  Technology      Copiers   599.970         5   
8123        19134     East  Technology      Copiers  1079.976         4   
8799        19134     East  Technology      Copiers  1799.970         5   
8808        19134     East  Technology       Phones   118.650         5   
9106        77041  Central  Technology     Machines   287.910         3   
9516        43402     E

####  There are 26 orders that have a discount greater than 30% and still earned a positive profit.

## ***Question No.7***

### 7- Find high-profit orders where Quantity > 5 and Category == 'Technology'.

## ***Answer No.7***

In [9]:
data[(data['Quantity'] > 5) & (data['Category'] == 'Technology')]

Ship Mode    Segment        Country           City           State  \
7     Standard Class   Consumer  United States    Los Angeles      California   
35       First Class  Corporate  United States     Richardson           Texas   
103   Standard Class   Consumer  United States         Aurora        Colorado   
106   Standard Class   Consumer  United States      Charlotte  North Carolina   
123      First Class   Consumer  United States     Wilmington        Delaware   
...              ...        ...            ...            ...             ...   
9840  Standard Class   Consumer  United States     Long Beach      California   
9882     First Class   Consumer  United States      Woodstock         Georgia   
9891    Second Class  Corporate  United States      Milwaukee       Wisconsin   
9895  Standard Class   Consumer  United States        Chicago        Illinois   
9941  Standard Class   Consumer  United States  San Francisco      California   

      Postal Code   Region    Category Sub-Category     Sales  Quantity  \
7           90032     West  Technology       Phones   907.152         6   
35          75080  Central  Technology       Phones  1097.544         7   
103         80013     West  Technology  Accessories   238.896         6   
106         28205    South  Technology  Accessories    74.112         8   
123         19805     East  Technology       Phones    68.040         7   
...           ...      ...         ...          ...       ...       ...   
9840        90805     West  Technology       Phones    44.736         8   
9882        30188    South  Technology  Accessories   272.610        13   
9891        53209  Central  Technology  Accessories   124.250         7   
9895        60623  Central  Technology  Accessories   153.824        11   
9941        94122     West  Technology  Accessories   223.580        14   

      Discount    Profit  
7          0.2   90.7152  
35         0.2  123.4737  
103        0.2  -26.8758  
106        0.2   17.6016  
123        0.0   19.7316  
...        ...       ...  
9840       0.2    4.4736  
9882       0.0   98.1396  
9891       0.0   48.4575  
9895       0.2   38.4560  
9941       0.0   87.1962  

[323 rows x 13 columns]

####  There are 323 high-profit orders in the Technology category with quantities greater than 5.

## ***Question No.8***

### 8- Select rows 0–4 and columns 'Ship Mode', 'Segment', 'Sales' by position using iloc

## ***Answer No.8***

####  There are 26 orders that have a discount greater than 30% and still earned a positive profit.

In [10]:
data.iloc[0:5, [0,1,9]]

Ship Mode    Segment     Sales
0    Second Class   Consumer  261.9600
1    Second Class   Consumer  731.9400
2    Second Class  Corporate   14.6200
3  Standard Class   Consumer  957.5775
4  Standard Class   Consumer   22.3680

## ***Question No.9***

### 9- Show Home Office segment orders with Sales > 1000 and Quantity > 3.

## ***Answer No.9***

In [11]:
data[(data['Segment'] == 'Home Office') & (data['Sales'] > 1000) & (data['Quantity'] > 3)]

Ship Mode      Segment        Country           City         State  \
244     Second Class  Home Office  United States      Lakeville     Minnesota   
247     Second Class  Home Office  United States      Lakeville     Minnesota   
353      First Class  Home Office  United States  New York City      New York   
463   Standard Class  Home Office  United States        Phoenix       Arizona   
515   Standard Class  Home Office  United States    Great Falls       Montana   
...              ...          ...            ...            ...           ...   
8890    Second Class  Home Office  United States       Columbia      Missouri   
8892  Standard Class  Home Office  United States      San Diego    California   
9056     First Class  Home Office  United States  New York City      New York   
9660  Standard Class  Home Office  United States  Santa Barbara    California   
9673  Standard Class  Home Office  United States   Philadelphia  Pennsylvania   

      Postal Code   Region         Category Sub-Category     Sales  Quantity  \
244         55044  Central        Furniture       Chairs  2001.860         7   
247         55044  Central  Office Supplies   Appliances  1503.250         5   
353         10009     East  Office Supplies      Binders  4355.168         4   
463         85023     West        Furniture       Tables  1272.630         6   
515         59405     West       Technology      Copiers  2999.950         5   
...           ...      ...              ...          ...       ...       ...   
8890        65203  Central  Office Supplies   Appliances  1927.590         7   
8892        92037     West        Furniture       Chairs  2676.672         9   
9056        10011     East        Furniture       Chairs  2621.322        11   
9660        93101     West       Technology       Phones  3023.928         9   
9673        19143     East       Technology       Phones  1499.970         5   

      Discount     Profit  
244        0.0   580.5394  
247        0.0   496.0725  
353        0.2  1415.4296  
463        0.5  -814.4832  
515        0.0  1379.9770  
...        ...        ...  
8890       0.0   751.7601  
8892       0.2   267.6672  
9056       0.1   553.3902  
9660       0.2   226.7946  
9673       0.4  -374.9925  

[73 rows x 13 columns]

####  There are 73 Home Office segment orders with sales exceeding 1000 and quantities greater than 3.

## ***Question No.10***

### 10- Find orders where Profit Margin > 0.3

Profit Margin = (Profit/Sales)

## ***Answer No.10***

In [12]:
data[(data['Profit'] / data['Sales']) > 0.3]

Ship Mode    Segment        Country           City           State  \
2       Second Class  Corporate  United States    Los Angeles      California   
8     Standard Class   Consumer  United States    Los Angeles      California   
12    Standard Class   Consumer  United States        Concord  North Carolina   
13    Standard Class   Consumer  United States        Seattle      Washington   
20      Second Class   Consumer  United States  San Francisco      California   
...              ...        ...            ...            ...             ...   
9982  Standard Class   Consumer  United States   Grand Rapids        Michigan   
9984  Standard Class   Consumer  United States     Long Beach        New York   
9986  Standard Class   Consumer  United States    Los Angeles      California   
9987  Standard Class  Corporate  United States         Athens         Georgia   
9992  Standard Class   Consumer  United States     Costa Mesa      California   

      Postal Code   Region         Category Sub-Category    Sales  Quantity  \
2           90036     West  Office Supplies       Labels   14.620         2   
8           90032     West  Office Supplies      Binders   18.504         3   
12          28027    South  Office Supplies        Paper   15.552         3   
13          98103     West  Office Supplies      Binders  407.976         3   
20          94109     West  Office Supplies      Binders   22.720         4   
...           ...      ...              ...          ...      ...       ...   
9982        49505  Central  Office Supplies        Paper   35.560         7   
9984        11561     East  Office Supplies       Labels   31.500        10   
9986        90008     West       Technology  Accessories   36.240         1   
9987        30605    South       Technology  Accessories   79.990         1   
9992        92627     West  Office Supplies        Paper   29.600         4   

      Discount    Profit  
2          0.0    6.8714  
8          0.2    5.7825  
12         0.2    5.4432  
13         0.2  132.5922  
20         0.2    7.3840  
...        ...       ...  
9982       0.0   16.7132  
9984       0.0   15.1200  
9986       0.0   15.2208  
9987       0.0   28.7964  
9992       0.0   13.3200  

[4145 rows x 13 columns]

####  There are 4145 orders with a profit margin greater than 30%

In [13]:
#Saving the data in 1 excel file with 10 Questions on different tabs.

In [14]:
with pd.ExcelWriter('Sameer_Assignment_12.xlsx') as writer:
    
    # Question No.1
    data[(data['Sales'] > 500) & (data['Quantity'] > 5)].to_excel(writer, sheet_name='Question 1', index=False)
    
    # Question No.2
    data[(data['Discount'] > 0.5) | (data['Profit'] < 1)].to_excel(writer, sheet_name='Question 2', index=False)
    
    # Question No.3
    data.loc[data['Region'] == 'East', ['Ship Mode','Profit']].to_excel(writer, sheet_name='Question 3', index=False)
    
    # Question No.4
    data[(data['Segment'] == 'Corporate') & (data['Region'] == 'Central')].to_excel(writer, sheet_name='Question 4', index=False)
    
    # Question No.5
    data[(data['Category'] == 'Furniture') & ~(data['Sub-Category'] == 'Chairs')].to_excel(writer, sheet_name='Question 5', index=False)
    
    # Question No.6
    data[(data['Discount'] > 0.3) & (data['Profit'] > 0)].to_excel(writer, sheet_name='Question 6', index=False)
    
    # Question No.7
    data[(data['Quantity'] > 5) & (data['Category'] == 'Technology')].to_excel(writer, sheet_name='Question 7', index=False)
    
    # Question No.8
    data.iloc[0:5, [0,1,9]].to_excel(writer, sheet_name='Question 8', index=False)
    
    # Question No.9
    data[(data['Segment'] == 'Home Office') & (data['Sales'] > 1000) & (data['Quantity'] > 3)].to_excel(writer, sheet_name='Question 9', index=False)
    
    # Question No.10
    data[(data['Profit'] / data['Sales']) > 0.3].to_excel(writer, sheet_name='Question 10', index=False)